lib/ABI/FunctionType.cpp

```diff
@@ -613,12 +650,98 @@ convertToCABIFunctionType(const model::RawFunctionType &Function,
 }
 
 std::optional<model::RawFunctionType>
-convertToRawFunctionType(const model::CABIFunctionType &Function,
-                         model::Binary &TheBinary) {
+tryConvertToRaw(const model::CABIFunctionType &Function,
+                model::Binary &TheBinary) {
   revng_assert(Function.ABI != model::ABI::Invalid);
   return skippingEnumSwitch<1>(Function.ABI, [&]<model::ABI::Values A>() {
     return ConvertionHelper<A>::toRaw(Function, TheBinary);
   });
 }
 
-} // namespace model
+Layout layout(const model::CABIFunctionType &Function) {
+  auto R = skippingEnumSwitch<1>(Function.ABI, [&]<model::ABI::Values A>() {
+    return ConvertionHelper<A>::getLayout(Function);
+  });
+  return Layout(std::move(R.Arguments),
+                std::move(R.ReturnValue),
+                std::move(R.CalleeSavedRegisters),
+                std::move(R.FinalStackOffset));
+}
+
+Layout layout(const model::RawFunctionType &Function) {
+  // Lay register arguments out.
+  std::decay<decltype(Layout::Arguments)>::type ResultArguments;
```

Comment:
This function heavily uses `std::decay` and `typename` for local vars, not very readable.
Either put the actual type or put some `using`s at the beginning.

Reply:
Gone: https://github.com/revng/revng/pull/131/commits/505909f2fca23b0a400bb6c17031fac8481b8472.